tools/model/apply/Main.cpp

```diff
@@ -0,0 +1,57 @@
+/// \file Main.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <string>
+
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/ToolOutputFile.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "revng/Model/ToolHelpers.h"
+#include "revng/Support/MetaAddress/YAMLTraits.h"
+#include "revng/Support/YAMLTraits.h"
+#include "revng/TupleTree/TupleTreeDiff.h"
+
+using namespace llvm;
+
+static cl::OptionCategory ThisToolCategory("Tool options", "");
+
+static cl::opt<std::string> PathModel(cl::Positional,
+                                      cl::cat(ThisToolCategory),
+                                      cl::desc("<input model>"),
+                                      cl::init("-"),
+                                      cl::value_desc("model"));
+
+static cl::opt<std::string> DiffPath(cl::Positional,
+                                     cl::cat(ThisToolCategory),
+                                     cl::desc("<model diff>"),
+                                     cl::value_desc("model"));
+
+static ModelOutputOptions<false> Options(ThisToolCategory);
+
+int main(int Argc, char *Argv[]) {
```

Comment:
I'd like `-R` (`man diff | grep -- -R`) and a roundtrip test to ensure that diffing and applying produces a result the same as the original one.

Reply:
how do you suggest to test the model without committing handwritten models for this purpose?

Reply:
Uhm, you can pick the `x86` `calc` artifact, extract the model, `sed s/functionname/otherfunctionname/`, compute the diff, apply, apply the opposite, test they are equal.
If you don't like `sed`, you can commit a patch.

Reply:
instead of using sed, which would only test changes of names, i extract the model from two unrelated binaries so it tests additions, changes and removals.